test/powershell/Language/Operators/TernaryOperator.Tests.ps1

```diff
@@ -0,0 +1,185 @@
+                @{ Script = '$env:123';  TokenKind = [TokenKind]::Variable; }
+                @{ Script = 'a?2:2';     TokenKind = [TokenKind]::Generic;  }
+                @{ Script = '1?2:3';     TokenKind = [TokenKind]::Generic;  }
+                @{ Script = 'a?';        TokenKind = [TokenKind]::Generic;  }
+                @{ Script = 'a?b';       TokenKind = [TokenKind]::Generic;  }
+                @{ Script = '1?';        TokenKind = [TokenKind]::Generic;  }
+                @{ Script = '?2:3';      TokenKind = [TokenKind]::Generic;  }
+            )
+
+            $testCases_incomplete = @(
+                @{ Script = '$true ?';     ErrorId = "ExpectedValueExpression";         AstType = [ErrorExpressionAst] }
+                @{ Script = '$true ? 3';   ErrorId = "MissingColonInTernaryExpression"; AstType = [ErrorExpressionAst] }
+                @{ Script = '$true ? 3 :'; ErrorId = "ExpectedValueExpression";         AstType = [TernaryExpressionAst] }
+            )
+        }
+
+        It "Question-mark and colon parsed correctly in '<Script>' when not in ternary expression context" -TestCases $testCases_basic {
+            param($Script, $TokenKind)
+
+            $tks = $null
+            $ers = $null
+            $result = [Parser]::ParseInput($Script, [ref]$tks, [ref]$ers)
+
+            $tks[0].Kind | Should -BeExactly $TokenKind
+            $tks[0].Text | Should -BeExactly $Script
+
+            if ($TokenKind -eq "Variable") {
+                $result.EndBlock.Statements[0].PipelineElements[0].Expression | Should -BeOfType 'System.Management.Automation.Language.VariableExpressionAst'
+                $result.EndBlock.Statements[0].PipelineElements[0].Expression.Extent.Text | Should -BeExactly $Script
+            } else {
+                $result.EndBlock.Statements[0].PipelineElements[0].CommandElements[0] | Should -BeOfType 'System.Management.Automation.Language.StringConstantExpressionAst'
+                $result.EndBlock.Statements[0].PipelineElements[0].CommandElements[0].Extent.Text | Should -BeExactly $Script
+            }
+        }
+
+        It "Question-mark and colon can be used as command names" {
+            function a?b:c { 'a?b:c' }
+            function 2?3:4 { '2?3:4' }
+
+            a?b:c | Should -BeExactly 'a?b:c'
+            2?3:4 | Should -BeExactly '2?3:4'
+        }
+
+        It "Generate incomplete parsing error properly for '<Script>'" -TestCases $testCases_incomplete {
+            param($Script, $ErrorId, $AstType)
+
+            $ers = $null
+            $result = [Parser]::ParseInput($Script, [ref]$null, [ref]$ers)
+
+            $ers.Count | Should -Be 1
+            $ers.IncompleteInput | Should -BeTrue
+            $ers.ErrorId | Should -BeExactly $ErrorId
+
+            $result.EndBlock.Statements[0].PipelineElements[0].Expression | Should -BeOfType $AstType
+        }
+
+        It "Generate ternary ast when possible" {
```

Comment:
The test purpose is to check when the operands are missing, we can still generate a ternary expression AST, with `ErrorExpressionAst` as the operands.
This would be useful in some tab completion scenarios.
The test case should have a better name and I will split them into two tests.